### PR TITLE
#15077 - Fix findFirst() return type

### LINKED ISSUE #15077: findFirst method return null, if record is not found
Hi,
i found that in phalcon4, when record is not found with findfirst method, this method retrun null, not false, like in phalcon3.
In offcial phalcon4 documentation its not mentioned. But if someoone looks in to phalcon .zep files, its ease to find, that findFirst can return mixed data type (because of Pahlcon/Mvc/Model/Query->execute() can https://github.com/phalcon/cphalcon/blob/fe17e3ae897b8d3f41d65ed328fcb734d1955b51/phalcon/Mvc/Model/Query.zep#L3643).

https://github.com/phalcon/ide-stubs/blob/d2adae3fba488ccba913a3f7fc310f23aa6de4a3/src/Phalcon/Mvc/Model.php#L791

### PR DESCRIPTION
**Pull request by Jeckerson**
Hello!

* Type: bug fix  | code quality
* Link to issue: #15077

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [ ] I wrote some tests for this PR
- [ ] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

### CODE REVIEW

File: phalcon/Mvc/Model.zep [hunk 1]
```diff
@@ -1534,7 +1534,7 @@ abstract class Model extends AbstractInjectionAware implements EntityInterface,
      *     'hydration' => null
      * ]
      */
-    public static function findFirst(var parameters = null) -> <ModelInterface> | bool
+    public static function findFirst(var parameters = null) -> <ModelInterface> | null
```
Comment (sergeyklay) [inline phalcon/Mvc/Model.zep:1537]: Please correct interface

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Review by sergeyklay — APPROVED** [review]
From design point of view I'm OK with this change. But I'm not sure about the target branch. Are we going to break backward compatibility in minor branch?

/сс @phalcon/core-team

**Comment by Jeckerson** [comment]
The problem, is that this needed to be changed back in `4.0.0`.

**Comment by ruudboon** [comment]
Why can't we keep it as null and update the docs?

**Comment by Jeckerson** [comment]
The docs are already updated.

**Comment by ruudboon** [comment]
Hmm now I see. We do break BC so if we approach it purely like that we should do it in v5. On the other hand I think it was intended to have it like this in v4 so you could see this as a bug that we need we need to fix and address it in v4.1

**Review by ruudboon — APPROVED** [review]
Technically it looks good to me. Let's wait for @niden before merging this one into 4.1.x

**Review by niden — APPROVED** [review]

**Pull request merged**